library/psa_crypto.c

```diff
@@ -5567,7 +5803,7 @@ static psa_status_t psa_generate_key_internal(
 #if defined(MBEDTLS_RSA_C) && defined(MBEDTLS_GENPRIME)
     if ( type == PSA_KEY_TYPE_RSA_KEY_PAIR )
     {
-        mbedtls_rsa_context *rsa;
+        mbedtls_rsa_context rsa;
```

Comment:
Why moving to an allocation on the stack? How do we know this won't be a problem on platforms with relatively small stacks? Otherwise, one could argue that this is a bit out of the scope of this commit/PR as well.

Reply:
The RSA context isn't that big, and all MPI members get dynamically allocated either way. I thought the code was cleaner this way without all the dynamic allocation... But OK, I'll update to put it back on the heap.

Relatively small stacks and RSA don't necessarily go together either...

Reply:
Thanks Gilles for providing some context here. Fine by me then.

Reply:
I've committed a changeset (a2371e5) that puts everything back on the heap and updates ecp and rsa to be consistent. Please look over and give feedback.